app/src/main/java/org/schabi/newpipe/player/helper/PlaybackParameterDialog.java

```diff
@@ -176,12 +201,47 @@ public Dialog onCreateDialog(@Nullable final Bundle savedInstanceState) {
     private void setupControlViews(@NonNull final View rootView) {
         setupHookingControl(rootView);
         setupSkipSilenceControl(rootView);
+        setupAdjustBySemitonesControl(rootView);
 
         setupTempoControl(rootView);
         setupPitchControl(rootView);
+        setupSemitoneControl(rootView);
+
+        togglePitchSliderType(rootView);
 
         setStepSize(stepSize);
-        setupStepSizeSelector(rootView);
+    }
+
+    private void togglePitchSliderType(@NonNull final View rootView) {
+
+        pitchControl = rootView.findViewById(R.id.pitchControl);
+        semitoneControl = rootView.findViewById(R.id.semitoneControl);
+
+        separatorStepSizeSelector = rootView.findViewById(R.id.separatorStepSizeSelector);
```

Comment:
Why are these two variables class fields if they are only used here? Make them local to this function (I know other parts of this file do it like you did, but that's not the way to go, it's just old code)